operator/cmd/syncclusterconfig/sync.go

```diff
@@ -413,16 +400,27 @@ func (s *Syncer) maybeMergeSuperusers(ctx context.Context, config map[string]any
 	if !ok {
 		// we have no superusers configuration, so don't do anything
 		logger.Info("Configuration does not contain a 'superusers' entry. Skipping superusers merge.")
-		return
+		return nil
 	}
 
-	superusersAny, ok := superusersConfig.([]any)
-	if !ok {
+	switch su := superusersConfig.(type) {
+	case []string:
+		superusers = append(superusers, su...)
+	case []any:
+		for _, s := range su {
+			if cast, ok := s.(string); ok {
+				superusers = append(superusers, cast)
+			} else {
+				logger.Info("Unable to cast value from %T to string, skipping.", s)
+			}
+		}
+	default:
 		logger.Info(fmt.Sprintf("Unable to cast superusers entry to array. Skipping superusers merge. Type is: %T", superusersConfig))
-		return
+		return fmt.Errorf("expected superusers entry to be an array of strings, got %T", superusersConfig)
```

Comment:
I don't think this should be an error case. It's less than ideal that superusers might be set incorrectly but I don't think that's a good enough reason to block the entire cluster configuration flow.

We could log this at the error level and/or increment a metric to improve the observability of this case?

Reply:
I've dropped the error. I'll add a ticket for the metric bit.